symengine/fields.h

```diff
@@ -0,0 +1,400 @@
+        // ! other is a just constant term
+        if (o_dict.rbegin()->first == 0) {
+            for (auto &arg : dict_) {
+                arg.second *= o_dict.rbegin()->second;
+                mp_fdiv_r(arg.second, arg.second, modulo_);
+            }
+            return static_cast<GaloisFieldDict &>(*this);
+        }
+
+        GaloisFieldDict res = GaloisFieldDict::mul(static_cast<GaloisFieldDict &>(*this),
+                                                   other);
+        res.dict_.swap(this->dict_);
+        return static_cast<GaloisFieldDict &>(*this);
+    }
+
+    template <class T>
+    friend GaloisFieldDict operator/(const GaloisFieldDict &a, const T &b)
+    {
+        GaloisFieldDict c = a;
+        c /= b;
+        return c;
+    }
+
+    GaloisFieldDict &operator/=(const integer_class &other)
+    {
+        if (other == integer_class(0)) {
+            throw std::runtime_error("ZeroDivisionError");
+        }
+        if (dict_.empty())
+            return static_cast<GaloisFieldDict &>(*this);
+        integer_class inv;
+        mp_invert(inv, other, modulo_);
+        for (auto &arg : dict_) {
+            arg.second *= inv;
+            mp_fdiv_r(arg.second, arg.second, modulo_);
+        }
+        return static_cast<GaloisFieldDict &>(*this);
+    }
+
+    GaloisFieldDict &operator/=(const GaloisFieldDict &other)
+    {
+        SYMENGINE_ASSERT(modulo_ == other.modulo_);
+        auto dict_divisor = other.get_dict();
+        if (dict_divisor.empty()) {
+            throw std::runtime_error("ZeroDivisionError");
+        }
+        if (dict_.empty())
+            return static_cast<GaloisFieldDict &>(*this);
+        integer_class inv;
+        mp_invert(inv, dict_divisor.rbegin()->second, modulo_);
+
+        // ! other is a just constant term
+        if (dict_divisor.rbegin()->first == 0) {
+            for (auto &iter : dict_) {
+                iter.second *= inv;
+                mp_fdiv_r(iter.second, iter.second, modulo_);
+            }
+            return static_cast<GaloisFieldDict &>(*this);
+        }
+        map_uint_mpz dict_out;
+        size_t deg_dividend = dict_.rbegin()->first;
+        size_t deg_divisor = dict_divisor.rbegin()->first;
+        if (deg_dividend < deg_divisor) {
+            dict_.clear();
+            return static_cast<GaloisFieldDict &>(*this);
+        }
+        dict_out = std::move(dict_);
+        dict_.clear();
+        integer_class coeff;
+        for (auto riter = dict_out.rbegin(); riter != dict_out.rend(); ++riter) {
+            if (riter->first < deg_divisor)
+                break;
+            size_t it = riter->first;
+            coeff = riter->second;
+            size_t lb = deg_divisor + it > deg_dividend ? 
+                        deg_divisor + it - deg_dividend : 0;
+            size_t ub = std::min(it + 1, deg_divisor);
+            for (size_t j = lb; j < ub; ++j) {
+                mp_addmul(coeff, dict_out[it - j + deg_divisor], -dict_divisor[j]);
```

Comment:
What I meant is you are iterating, dict_divisor from `lb` to `ub` via `dict_divisor[j]`. Can't this be done by iterating `dict_divisor` ?